### ISSUE TITLE
Lua Script Error: moveUpstairs

### ISSUE THREAD

**Issue by omarcopires** [otland/forgottenserver#4316]
### Before creating an issue, please ensure:
- [x] This is a bug in the software that resides in this repository, and not a support matter (use https://otland.net/forums/support.16/ for support)
- [x] This issue is reproducible without changes to the C++ code in this repository

### Steps to reproduce (include any configuration/script required to reproduce)
1. Use a ladder to climb

### Expected behaviour 
Climb the ladder and return no error.

### Actual behaviour 
It climbs the ladder and returns an error to the console
```Lua Script Error: [Action Interface]
data/actions/scripts/others/teleport.lua:onUse
LuaScriptInterface::getNumber(). Argument -1 has out-of-range value for unsigned short: -1
stack traceback:
        data/lib/core/position.lua:38: in function 'moveUpstairs'
        data/actions/scripts/others/teleport.lua:4: in function <data/actions/scripts/others/teleport.lua:2>```

**Comment by omarcopires** [comment]
> Cannot reproduce, do I need some special ladder or can it be simple one?

They aren't all stairs, I don't know exactly how to reproduce them, but in my case it's a staircase on the 3rd floor that leads to the 4th. However, there are other stairs on the same floor that don't return any errors.

**Comment by projectMWX** [comment]
> @omarcopires Cannot reproduce, do I need some special ladder or can it be simple one? ![ladder](https://user-images.githubusercontent.com/95135755/219767719-de88f7d8-b537-4f74-a20b-df25f82425ab.gif) [ ![ladder](https://user-images.githubusercontent.com/95135755/219767719-de88f7d8-b537-4f74-a20b-df25f82425ab.gif) ](https://user-images.githubusercontent.com/95135755/219767719-de88f7d8-b537-4f74-a20b-df25f82425ab.gif) [ ](https://user-images.githubusercontent.com/95135755/219767719-de88f7d8-b537-4f74-a20b-df25f82425ab.gif)

It only occurs when the default south tile is not walkable or doesn't exist, and therefore loops through all neighbour tiles:
https://github.com/otland/forgottenserver/blob/master/data/lib/core/position.lua#L32

**Comment by projectMWX** [comment]
Exactly. The magic method for the addition only for applies for Position + Position, and the default South check works as the x, y and z are all unsigned values, but anything north or northwest includes signed values and therefore throws an error.

As you can see in this PR, best to just apply each value rather than use the magic method:
https://github.com/otland/forgottenserver/pull/4323
